client/landing/gutenboarding/onboarding-block/edit.tsx

```diff
@@ -142,6 +189,12 @@ const OnboardingEdit: React.FunctionComponent< BlockEditProps< Attributes > > =
 		}
 	}, [ location, step, setLastLocation ] );
 
+	const styleStepIfNotSkipped = shouldSkipStyleStep() ? (
+		redirectToDesignWithoutFontsStep
+	) : (
+		<StylePreview />
+	);
```

Comment:
This logic is used when navigating directly to the `style` step (e.g. typing `/new/style` in the browser's bar).

The idea is: if the user has selected a design, but that design doesn't have fonts specified, then the `style` step should be skipped by redirecting to the step that would normally come immediately after (this is determined in the `getDesignWithoutFontsPath ` function).

The